app/client/jsx/PuckSelect.jsx

```diff
@@ -1,108 +1,98 @@
-import React, { Component } from 'react';
-import PuckBox from './PuckBox.jsx';
+import React, { Component } from 'react'
+import PuckBox from './PuckBox.jsx'
 import { Avatars } from './avatars.jsx'
 
-
 class PuckSelect extends Component {
-    constructor(props) {
-        super(props)
-        this.state = {
-            columnOpen: false,
-            rowOpen: null,
-            avatarDesign: '',
-            avatarColorway: '',
-            opacity: this.props.opacity
-        }
-        this.onButtonClick = this.onButtonClick.bind(this)
-        this.onButtonClickReset = this.onButtonClickReset.bind(this)
-        this.handleClickDesign = this.handleClickDesign.bind(this)
-        this.handleClickColor = this.handleClickColor.bind(this)
-      }
+  constructor (props) {
+    super(props)
+    this.initialState = {
+      columnOpen: false,
+      rowOpen: null,
+      avatarDesign: '',
+      avatarColorway: '',
+      opacity: this.props.opacity
+    }
+    this.state = this.initialState
+  }
 
-  onButtonClick() {
-    this.setState({ columnOpen: true }) }
+  handleButtonClick () { this.setState({ columnOpen: true }) }
 
-  onButtonClickReset() {
-    this.setState({ columnOpen: false })
-    this.setState({ rowOpen: null })
-    this.setState({ avatarDesign: '' })
-    this.setState({ avatarColorway: '' })
+  handleButtonClickReset () {
+    this.setState(this.initialState)
     this.setState({ columnOpen: true })
   }
-  handleClickDesign(key, index) {
-    this.setState({ avatarDesign: key, rowOpen: index }) }
-  handleClickColor(variantKey) {
+
+  handleClickDesign (key, index) {
+    this.setState({ avatarDesign: key, rowOpen: index })
+  }
+
+  handleClickColor (variantKey) {
     this.setState({ avatarColorway: variantKey })
     this.props.handleSelect({
       type: this.state.avatarDesign,
       color: variantKey
     })
   }
 
-  render() {
+  render () {
+    console.log = console.warn = console.error = () => {}
```

Comment:
this looks mighty suspicious. won't this leak out and affect the function globally?

Reply:
I'd agree that it's suspicious, even distasteful. Suffice it to say it only entered the picture after many involved attempts to determine the cause, and at Morgane's suggestion.

This said, if it's cutting across scope we've got a problem

Reply:
personally i'd much rather have some garbage in the console than to clobber this function globally

Reply:
what is it that this is trying to solve? I forget

Reply:
one of these. of unprecedented tenacity 
![Screen Shot 2020-05-19 at 11 55 12 AM](https://user-images.githubusercontent.com/37454737/82349068-a7254280-99c7-11ea-85d9-78238d688b79.png)

Reply:
I think you need to set a unique key on all the elements inside PuckBox, maybe
https://stackoverflow.com/questions/28329382/understanding-unique-keys-for-array-children-in-react-js

Reply:
I suggested silencing it. You can silence it with a try / catch

Reply:
@seangibat it seems like a more intractable problem than that for whatever reason

Reply:
It's tricky to make sure each div has a unique key, but if you do that, you won't get the warning. I tried it here and it worked: https://github.com/morganecf/jitsi-party/pull/116 (But this is really messy demonstration and it might be that not all of it is necessary.)

Reply:
On further testing I realized the only thing you need to do is set the key on the main outer div! I don't quite understand why.

Reply:
I'll defer to @danielmdavis on this one! That fix looks good to me.